### PR TITLE
LW-7438 APY job

### PR DESCRIPTION
**Pull request by iccicci**
# Context

We need to re-enable the ROS (once called APY) in our stake pool metrics.

# Proposed Solution

- Added a _stake pool reward_ job scheduled on epoch rollover.

# Important Changes Introduced

_Renamed_ the properties for new ROS naming: actually new `ros` related properties was added while `apy` related properties was not removed to avoid a huge impact on `DbSyncStakePoolProvider`; better to remove these properties while removing the db-sync provider.

### CODE REVIEW

File: packages/projection-typeorm/src/operators/storeStakePoolRewardsJob.ts
Comment (mkazlauskas) [inline packages/projection-typeorm/src/operators/storeStakePoolRewardsJob.ts:17]:
```suggestion
      pgBoss.send(STAKE_POOL_REWARDS, { epochNo }, { ...defaultJobOptions, singletonKey: `pool-rewards-${epochNo}`, slot });
```

Comment (iccicci) [reply packages/projection-typeorm/src/operators/storeStakePoolRewardsJob.ts]: a5a76e24 should solve this

File: packages/core/src/Cardano/types/StakePool/StakePool.ts
Comment (mkazlauskas) [inline packages/core/src/Cardano/types/StakePool/StakePool.ts:7]: There are a lot of formatting changes for doc comments. I'm fine with new format, but it would be best to do it in a separate commit:
- enforce it via our tooling (probably eslint)
- apply the new format to entire codebase

File: packages/core/src/Cardano/types/StakePool/StakePool.ts
Comment (mkazlauskas) [inline packages/core/src/Cardano/types/StakePool/StakePool.ts:101]: This is a breaking change, please reflect in the commit message

File: packages/core/src/Cardano/types/StakePool/StakePool.ts
Comment (mkazlauskas) [inline packages/core/src/Cardano/types/StakePool/StakePool.ts:124]: This is a breaking change, please reflect in the commit message

File: packages/core/src/util/slotCalc.ts
Comment (mkazlauskas) [inline packages/core/src/util/slotCalc.ts:156]: This is a breaking change, please reflect in the commit message

File: packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts
Comment (mkazlauskas) [inline packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts:155]: Shouldn't this behave the same as `case 'apy'`?

File: packages/projection-typeorm/src/entity/CurrentPoolMetrics.entity.ts
Comment (mkazlauskas) [inline packages/projection-typeorm/src/entity/CurrentPoolMetrics.entity.ts:21]: We should probably have a cardano-services migration too in this commit

File: packages/cardano-services/src/Projection/prepareTypeormProjection.ts
Comment (mkazlauskas) [inline packages/cardano-services/src/Projection/prepareTypeormProjection.ts:50]: We should probably add a migration adding the required table in this commit

File: packages/e2e/test/providers/StakePoolProvider.test.ts
Comment (mkazlauskas) [inline packages/e2e/test/providers/StakePoolProvider.test.ts:425]: This should probably be part of the commit that updates the provider in cardano-services

File: packages/cardano-services/test/jest-setup/snapshots/asset.sql
Comment (mkazlauskas) [inline packages/cardano-services/test/jest-setup/snapshots/asset.sql:1]: Nice, generating this is part of LW-8695
Although commits probably have to be reordered, as some tests in previous commits of cardano-services are probably relying on changes of re-generated data.

**Review by mkazlauskas — CHANGES_REQUESTED** [review]

File: packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts
Comment (iccicci) [reply packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts:155]: Actually yes, but does this worth?

Planning for a db-sync drop I spent no time on `DbSyncStakePoolProvider` (but the not implemented error).

File: packages/projection-typeorm/src/entity/CurrentPoolMetrics.entity.ts
Comment (iccicci) [reply packages/projection-typeorm/src/entity/CurrentPoolMetrics.entity.ts:21]: Yes sure!

I'll add migrations once all the changes on schema will be reviewed / approved to add them only once rather than applying them now with the risk of need to change them during the review process (with the relative higher developer time cost).

File: packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts
Comment (mkazlauskas) [reply packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts:155]: Alright let's not spend time on this

File: packages/core/src/Cardano/types/StakePool/StakePool.ts
Comment (iccicci) [reply packages/core/src/Cardano/types/StakePool/StakePool.ts:7]: Once merged #959 , simply rebasing this branch should solve this as well

Comment (iccicci) [reply packages/core/src/Cardano/types/StakePool/StakePool.ts]: 9bb6579a4 should solve this

File: packages/core/src/Cardano/types/StakePool/StakePool.ts
Comment (iccicci) [reply packages/core/src/Cardano/types/StakePool/StakePool.ts:99]: 9bb6579a4 should solve this

File: packages/core/src/Cardano/types/StakePool/StakePool.ts
Comment (iccicci) [reply packages/core/src/Cardano/types/StakePool/StakePool.ts:122]: 9bb6579a4 should solve this

File: packages/projection-typeorm/src/entity/CurrentPoolMetrics.entity.ts
Comment (iccicci) [reply packages/projection-typeorm/src/entity/CurrentPoolMetrics.entity.ts:21]: 1b2570abe should solve this

File: packages/core/src/util/slotCalc.ts
Comment (iccicci) [reply packages/core/src/util/slotCalc.ts:156]: db4d771dd should solve this

File: packages/e2e/test/providers/StakePoolProvider.test.ts
Comment (iccicci) [reply packages/e2e/test/providers/StakePoolProvider.test.ts:425]: Squashed in b4b1ca188

File: packages/core/src/util/slotCalc.ts
Comment (iccicci) [reply packages/core/src/util/slotCalc.ts:156]: db4d771dd should solve this

File: packages/projection-typeorm/src/entity/CurrentPoolMetrics.entity.ts
Comment (iccicci) [reply packages/projection-typeorm/src/entity/CurrentPoolMetrics.entity.ts:21]: 1b2570abe should solve this

File: packages/cardano-services/src/Projection/prepareTypeormProjection.ts
Comment (iccicci) [reply packages/cardano-services/src/Projection/prepareTypeormProjection.ts:47]: 5303986e3 should solve this

File: packages/cardano-services/test/jest-setup/snapshots/asset.sql
Comment (iccicci) [reply packages/cardano-services/test/jest-setup/snapshots/asset.sql:1]: Moving 77aff56ba earlier in commits history should solve this

File: packages/core/src/Cardano/types/StakePool/StakePool.ts
Comment (mkazlauskas) [reply packages/core/src/Cardano/types/StakePool/StakePool.ts:99]: The commit message is now a little confusing in terms of CHANGELOG, as it will say sth like:
Breaking changes:
- core: prepares the types for typeorm stake pool provider w…

A more accurate commit message would be something like that:
```
refactor(core)!: prepares the types for typeorm stake pool provider w…

BREAKING CHANGE: deprecated StakePoolEpochRewards.epoch
- deprecated StakePoolEpochRewards.memberROI
```

File: packages/core/src/util/slotCalc.ts
Comment (mkazlauskas) [reply packages/core/src/util/slotCalc.ts:156]: Same as https://github.com/input-output-hk/cardano-js-sdk/pull/882#discussion_r1371182775

File: packages/projection-typeorm/src/entity/CurrentPoolMetrics.entity.ts
Comment (mkazlauskas) [reply packages/projection-typeorm/src/entity/CurrentPoolMetrics.entity.ts:21]: Probably doesn't matter for this table, but `RENAME COLUMN column_name TO new_column_name;` would preserve any existing values

File: packages/core/src/Cardano/types/StakePool/StakePool.ts
Comment (iccicci) [reply packages/core/src/Cardano/types/StakePool/StakePool.ts:99]: d476e6e2d should solve this

File: packages/core/src/util/slotCalc.ts
Comment (iccicci) [reply packages/core/src/util/slotCalc.ts:156]: 2a19704cc should solve this

File: packages/core/src/Cardano/types/StakePool/StakePool.ts
Comment (mkazlauskas) [reply packages/core/src/Cardano/types/StakePool/StakePool.ts:99]: This formatting will result in a change log like:
```
Breaking changes:
- - Deprecated QueryStakePoolsArgs.apyEpochsBackLimit
- ...
```

Because the 1st breaking change must be inline :sweat_smile:

Comment (iccicci) [reply packages/core/src/Cardano/types/StakePool/StakePool.ts:99]: f79314b98 should solve this

**Review by mkazlauskas — APPROVED** [review]
Great work :rocket: :muscle:

**Comment by VanessaPC** [comment]
I think it looks great! 💪

**Review by VanessaPC — APPROVED** [review]

**Pull request merged**
